libraries/AP_Mount/AP_Mount_SToRM32.cpp

```diff
@@ -59,31 +57,39 @@ void AP_Mount_SToRM32::update()
         // RC radio manual angle control, but with stabilization from the AHRS
         case MAV_MOUNT_MODE_RC_TARGETING: {
             // update targets using pilot's RC inputs
-            MountTarget rc_target {};
-            if (get_rc_rate_target(rc_target)) {
-                update_angle_target_from_rate(rc_target, _angle_rad);
-            } else if (get_rc_angle_target(rc_target)) {
-                _angle_rad = rc_target;
+            if (get_rc_rate_target(mnt_target.rate_rads)) {
+                mnt_target.target_type = MountTargetType::RATE;
+                update_angle_target_from_rate(mnt_target.rate_rads, mnt_target.angle_rad);
+            } else {
```

Comment:
I think then call to, "get_rc_angle_target" should be on this line.  If the calls to both "get_rc_rate_target" and "get_rc_angle_target" fail then no state should be changed.